src/lib.rs

```diff
@@ -1116,3 +1177,29 @@ impl<'ui> Ui<'ui> {
         unsafe { sys::igIsItemHovered(ImGuiHoveredFlags::empty()) }
     }
 }
+
+/// # Draw list for custom drawing
+impl<'ui> Ui<'ui> {
+    /// Get access to drawing API
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// # use imgui::*;
+    /// fn custom_draw(ui: &Ui) {
+    ///     ui.with_window_draw_list(|draw_list| {
+    ///         // Draw a line
+    ///         const WHITE: [f32; 3] = [1.0, 1.0, 1.0];
+    ///         draw_list.add_line([100.0, 100.0], [200.0, 200.0], WHITE).build();
+    ///         // Continue drawing ...
+    ///     });
+    /// }
+    /// ```
+    pub fn with_window_draw_list<F>(&self, f: F)
```

Comment:
Using the `with_` + closure method here feels strange, because there's no cleanup function that needs to be called. Also, this doesn't add any extra safety because it can be called recursively multiple times. `WindowDrawList::new` is also public, so this is not the only way to construct a `WindowDrawList`

Reply:
Sure enough. I propose then to rename the function to `get_window_draw_list` and have it return a `WindowDrawList` object.

As you said, `WindowDrawList::new` should only be public within the crate. 
cc878b4  fixed this issue.